First iteration of a job system

Remscar opened the pull request:
This is a first iteration of a job system.

A "Job" is a collection of "Abilities."

Players can have only one "Job" but based on that job will have access to a set of "Abilities."

chrisspieler commented on code/jobs/Jobs.cs:
The code here may be clearer if JobAbilities had FlagsAttribute. This would allow for JobDetails.Abilities and PlayerJob.Abilities to be of the JobAbilities type. Then, PlayerJob.HasAbility could be made more readable by using Enum.HasFlag.

chrisspieler commented on code/player/Player.Jobs.cs:
```diff
@@ -0,0 +1,24 @@
+    [BindComponent]
+    public Jobs.PlayerJob Job { get; }
+
+    [ConCmd.Server("cinema.jobtest")]
+    public static void OnReload()
+    {
+        var ply = ConsoleSystem.Caller.Pawn as Player;
+        //ply.Components.Add(Jobs.PlayerJob.CreateFromDetails(Jobs.JobDetails.All.First()));
+    }
```
```suggestion

```
Seems unused.

Remscar replied:
Thanks for catching this

chrisspieler commented on code/game/Game.Debug.cs:
```diff
@@ -166,4 +167,41 @@ public static void SpawnNpcCMD(string npcName)
+    {
+        if (ConsoleSystem.Caller.Pawn is not Player player) return;
+        Log.Info($"Job: {player.Job.JobDetails.Name}");
+        Log.Info($"Abilities: {player.Job.JobDetails.Abilities.ToNiceBinary()}");
```
If JobDetails.Abilities were of type JobAbilities as suggested in my comment on Jobs.cs, there would be an easier way to enumerate job abilities here, making the ToNiceBinary extension obsolete. You'd also get access to human-readable names.

chrisspieler reviewed (commented):
Looks good for the most part, but I think there may be a missed opportunity in not using FlagsAttribute, and there's some seemingly unused code.

Remscar commented:
> Looks good for the most part, but I think there may be a missed opportunity in not using FlagsAttribute, and there's some seemingly unused code.

Excellent feedback, thank you!

I've adjusted the code to use `[Flag]` and removed the un-used code

chrisspieler reviewed (approved):
Looks good to me
